vllm/engine/arg_utils.py

```diff
@@ -418,6 +418,8 @@ class EngineArgs:
     kv_transfer_config: Optional[KVTransferConfig] = None
     kv_events_config: Optional[KVEventsConfig] = None
 
+    reasoning_config: Optional[ReasoningConfig] = None
```

Comment:
This might be better to avoid `Optional`:

```suggestion
    reasoning_config: ReasoningConfig = get_field(VllmConfig, "reasoning_config")
```